pkg/healthcheck/healthcheck_test.go

```diff
@@ -0,0 +1,355 @@
+		})
+	})
+	var _ = Context("NewTcpDialer() should", func() {
+		var _ = It("should use dialer with 3s timeout", func() {
+			c := fake.NewSimpleClientset()
+			hc, err := NewHealthChecker(c, NewHealthCheckToolkit(nil, fakeUpGetByName, NewTcpDialer("")), nil)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(hc).ToNot(BeNil())
+			Expect(hc.IsInitialized()).To(BeFalse())
+			d := hc.toolkit.tcpDialer.(*net.Dialer)
+			Expect(d.Timeout).To(Equal(time.Second * 3))
+		})
+		var _ = It("should use dialer with 5s timeout", func() {
+			c := fake.NewSimpleClientset()
+			hc, err := NewHealthChecker(c, NewHealthCheckToolkit(nil, fakeUpGetByName, NewTcpDialer("5")), nil)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(hc).ToNot(BeNil())
+			Expect(hc.IsInitialized()).To(BeFalse())
+			d := hc.toolkit.tcpDialer.(*net.Dialer)
+			Expect(d.Timeout).To(Equal(time.Second * 5))
+		})
+		var _ = It("should use dialer with 500ms timeout", func() {
+			c := fake.NewSimpleClientset()
+			hc, err := NewHealthChecker(c, NewHealthCheckToolkit(nil, fakeUpGetByName, NewTcpDialer("500ms")), nil)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(hc).ToNot(BeNil())
+			Expect(hc.IsInitialized()).To(BeFalse())
+			d := hc.toolkit.tcpDialer.(*net.Dialer)
+			Expect(d.Timeout).To(Equal(time.Millisecond * 500))
+		})
+	})
+	var _ = Context("CheckReachability() should", func() {
+		var _ = It("should return error if cannot reach host", func() {
+			c := fake.NewSimpleClientset()
+			nc := &NetHealthcheckConfig{
+				Reachability: []NetReachabilityItem{{Host: "someHost", Port: 42}},
+			}
+			hc, err := NewHealthChecker(c, NewHealthCheckToolkit(nil, fakeUpGetByName, &fakeDialerError{}), nc)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(hc).ToNot(BeNil())
+			Expect(hc.IsInitialized()).To(BeFalse())
+			err = hc.CheckReachability()
+			Expect(err).To(HaveOccurred())
+		})
+		var _ = It("should return no error if can reach host", func() {
+			c := fake.NewSimpleClientset()
+			nc := &NetHealthcheckConfig{
+				Reachability: []NetReachabilityItem{{Host: "someHost", Port: 42}},
+			}
+			hc, err := NewHealthChecker(c, NewHealthCheckToolkit(nil, fakeUpGetByName, &fakeDialer{}), nc)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(hc).ToNot(BeNil())
+			Expect(hc.IsInitialized()).To(BeFalse())
+			err = hc.CheckReachability()
+			Expect(err).ToNot(HaveOccurred())
+		})
+		var _ = It("should return no error if can reach host but connection was refused", func() {
+			c := fake.NewSimpleClientset()
+			nc := &NetHealthcheckConfig{
+				Reachability: []NetReachabilityItem{{Host: "someHost", Port: 42}},
+			}
+			hc, err := NewHealthChecker(c, NewHealthCheckToolkit(nil, fakeUpGetByName, &fakeDialerRefused{}), nc)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(hc).ToNot(BeNil())
+			Expect(hc.IsInitialized()).To(BeFalse())
+			err = hc.CheckReachability()
+			Expect(err).ToNot(HaveOccurred())
+		})
+	})
+})
+
+type fakeFRRManagerError struct{}
```

Comment:
instead of writing mocks yourself, you could also have a look at https://github.com/uber-go/mock

It's pretty nice, and allows to register handlers to return custom values for testing and so on, so you only need to generate one mock and can use it for everything. Might be a bit overkill here though.

Reply:
OK, I've removed my mocks and uber-go/mock is now used.

Reply:
Would be good to add a `// go:generate <gomock command>` somewhere, maybe in proximity of those interfaces. Then we can just run `go generate ./...` to regenerate if necessary

Reply:
Done